lib/cart/apis/cart_api.dart

```diff
@@ -8,15 +9,39 @@ const PATH = '/occ/v2/electronics-spa/users/anonymous/carts/';
 const PARAMS =
     '?fields=DEFAULT,potentialProductPromotions,appliedProductPromotions,potentialOrderPromotions,appliedOrderPromotions,entries(totalPrice(formattedValue),product(images(FULL),stock(FULL)),basePrice(formattedValue,value),updateable),totalPrice(formattedValue),totalItems,totalPriceWithTax(formattedValue),totalDiscounts(value,formattedValue),subTotal(formattedValue),totalUnitCount,deliveryItemsQuantity,deliveryCost(formattedValue),totalTax(formattedValue,%20value),pickupItemsQuantity,net,appliedVouchers,productDiscounts(formattedValue),user,saveTime,name,description&lang=en&curr=USD';
 
+const NEW_CART_PARAMS =
+    'DEFAULT,potentialProductPromotions,appliedProductPromotions,potentialOrderPromotions,appliedOrderPromotions,entries(totalPrice(formattedValue),product(images(FULL),stock(FULL)),basePrice(formattedValue,value),updateable),totalPrice(formattedValue),totalItems,totalPriceWithTax(formattedValue),totalDiscounts(value,formattedValue),subTotal(formattedValue),totalUnitCount,deliveryItemsQuantity,deliveryCost(formattedValue),totalTax(formattedValue,%20value),pickupItemsQuantity,net,appliedVouchers,productDiscounts(formattedValue),user&lang=en&curr=USD';
+
 class CartApi {
   final http = Dio();
 
   Future<Cart> fetchCart({String? cartCode}) async {
-    final response = await GetIt.instance
-        .get<Dio>(instanceName: Singletons.hybrisClient)
-        .get('${dotenv.get('HYBRIS_BASE_URL')}$PATH$cartCode$PARAMS');
+    try {
+      final response = await GetIt.instance
+          .get<Dio>(instanceName: Singletons.hybrisClient)
+          .get('${dotenv.get('HYBRIS_BASE_URL')}$PATH$cartCode$PARAMS');
+
+      return Cart.fromJson(response.data);
+    } on DioException catch (e) {
+      // try to deserialize error response
+      if (e.response?.data != null) {
+        throw CartException.fromJson(e.response!.data['errors'].first);
+      }
+
+      rethrow;
+    }
+  }
+
+  Future<Cart> createCart() async {
+    try {
+      final response = await GetIt.instance
+          .get<Dio>(instanceName: Singletons.hybrisClient)
+          .post('${dotenv.get('HYBRIS_BASE_URL')}$PATH$PARAMS', data: {});
 
-    return Cart.fromJson(response.data);
+      return Cart.fromJson(response.data);
+    } on DioException catch (_) {
+      rethrow;
```

Comment:
No try/catch for now if we follow the principle of "dealing with exception when we actually face them".

Or we can keep the try/catch and copy-paste the parsing code from `fetchCart`, we can safely assume it will require the same logic

Reply:
Let's leave it off for now and deal when we face it (or learn more about hybris possible exceptions)